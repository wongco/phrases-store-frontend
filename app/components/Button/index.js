/* eslint-disable indent */
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Button = styled.button`
  width: 5em;
  border: 1px solid ${({ borderColor }) => borderColor || 'black'};
  color: ${({ color, disabled }) => (!disabled && color) || 'gray'};
  background-color: ${({ bgColor, disabled }) => !disabled && bgColor};

  &:hover {
    background-color: ${({ hoverBgColor, disabled }) =>
      !disabled && hoverBgColor};
    color: ${({ hoverColor, disabled }) => !disabled && hoverColor};
  }
`;

Button.propTypes = {
  disabled: PropTypes.bool,
  color: PropTypes.string,
  borderColor: PropTypes.string,
  hoverBgColor: PropTypes.string,
  hoverColor: PropTypes.string,
};

export default Button;
