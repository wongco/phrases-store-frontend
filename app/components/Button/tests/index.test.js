import React from 'react';
import { shallow, mount } from 'enzyme';
import { enzymeFind } from 'styled-components/test-utils';
import toJson from 'enzyme-to-json';
import Button from '../index';

describe('<Button />', () => {
  it('renders without crashing', () => {
    shallow(<Button />);
  });

  it('matches snapshot for default criteria', () => {
    const wrapper = shallow(<Button />);
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });

  it('should render a prop', () => {
    const disabled = true;
    const color = 'blue';
    const wrapper = mount(<Button disabled color="blue" />);
    const renderedComponent = enzymeFind(wrapper, Button);
    expect(renderedComponent.prop('disabled')).toBe(disabled);
    expect(renderedComponent.prop('color')).toBe(color);
  });

  it('should render its text', () => {
    const children = 'Text';
    const wrapper = mount(<Button>{children}</Button>);
    const renderedComponent = enzymeFind(wrapper, Button);
    expect(renderedComponent.contains(children)).toBe(true);
  });
});
