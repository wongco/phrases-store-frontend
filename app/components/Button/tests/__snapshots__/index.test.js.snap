// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> matches snapshot for default criteria 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "Button-sc-1fkoyq4-0",
        "isStatic": false,
        "rules": Array [
          "width:5em;border:1px solid ",
          [Function],
          ";color:",
          [Function],
          ";background-color:",
          [Function],
          ";&:hover{background-color:",
          [Function],
          ";color:",
          [Function],
          ";}",
        ],
      },
      "displayName": "Button",
      "propTypes": Object {
        "borderColor": [Function],
        "color": [Function],
        "disabled": [Function],
        "hoverBgColor": [Function],
        "hoverColor": [Function],
      },
      "render": [Function],
      "styledComponentId": "Button-sc-1fkoyq4-0",
      "target": "button",
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
